src/main/java/cz/smarteon/loxone/app/state/ControlState.java

```diff
@@ -0,0 +1,54 @@
+package cz.smarteon.loxone.app.state;
+
+import cz.smarteon.loxone.LoxoneWebSocket;
+import cz.smarteon.loxone.app.model.Control;
+import cz.smarteon.loxone.message.ControlCommand;
+import cz.smarteon.loxone.message.TextEvent;
+import cz.smarteon.loxone.message.ValueEvent;
+import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Base class for all the controlStates in loxone application
+ */
+@RequiredArgsConstructor
+public abstract class ControlState<T extends Control> {
+
+    /**
+     * The webSocket connection to communicate with the loxone miniserver.
+     */
+    protected final LoxoneWebSocket webSocket;
```

Comment:
I wonder whether we should rather make it all dependent on Loxone class - as single dependency?

Reply:
Okay, makes sense, if the internals of the Loxone class changes we can hide away the changes for the ControlState classes